src/main/java/com/danielfrak/code/keycloak/providers/rest/LegacyProvider.java

```diff
@@ -108,6 +108,7 @@ public boolean updateCredential(RealmModel realm, UserModel user, CredentialInpu
         String link = user.getFederationLink();
         if (link != null && !link.isBlank()) {
             user.setFederationLink(null);
+            legacyUserService.removeByUsername(user.getUsername());
```

Comment:
I see `ConfigurationProperties.USE_USER_ID_FOR_CREDENTIAL_VERIFICATION` earlier. I wonder if we should use that same toggle to chose between `removeByUsername` vs `removeByEmail`.